lib/mongo/grid/stream/read.rb

```diff
@@ -167,7 +167,8 @@ def ensure_readable!
           end
 
           def view
-            @view ||= fs.chunks_collection.find({ :files_id => file_id }, options).read(read_preference).sort(:n => 1)
+            @view ||= (opts = options.merge(read: read_preference) if read_preference
```

Comment:
I'm having a little trouble parsing exactly what this is doing.  Is line 171 a separate statement being grouped with the previous one for the purpose of having its return value stored in view? If so, I think it might be clearer to move the assignment of `opts` above the line 170 and then change line 170 to `@view ||= fs.chunks_collection.find(...)`.